presto-spi/src/main/java/com/facebook/presto/spi/Page.java

```diff
@@ -238,11 +238,12 @@ public void compact()
      * This allows streaming data sources to skip sections that are not
      * accessed in a query.
      */
-    public void assureLoaded()
+    public Page assureLoaded()
     {
-        for (Block block : blocks) {
-            block.assureLoaded();
-        }
+        return new Page(
+                Arrays.stream(blocks)
```

Comment:
Avoid streams in execution code path.

also, the cmt msg says `Make Block.assureLoaded() ...`. Mention `Page` as well.